client/api/volume/v1beta3/api.proto

```diff
@@ -5,128 +5,129 @@ package v1beta3;
-    // all volumes on a Disk device
+    // ListVolumesOnDisk returns the volume IDs (in \\.\Volume{GUID} format) for all volumes on a Disk device.
     rpc ListVolumesOnDisk(ListVolumesOnDiskRequest) returns (ListVolumesOnDiskResponse) {}
-    // MountVolume mounts the volume at the requested global staging path
+    // MountVolume mounts the volume at the requested global staging path.
     rpc MountVolume(MountVolumeRequest) returns (MountVolumeResponse) {}
-    // DismountVolume gracefully dismounts a volume
-    rpc DismountVolume(DismountVolumeRequest) returns (DismountVolumeResponse) {}
-    // IsVolumeFormatted checks if a volume is formatted with NTFS
+    // UnmountVolume gracefully unmounts a volume.
+    rpc UnmountVolume(UnmountVolumeRequest) returns (UnmountVolumeResponse) {}
+    // IsVolumeFormatted checks if a volume is formatted.
     rpc IsVolumeFormatted(IsVolumeFormattedRequest) returns (IsVolumeFormattedResponse) {}
-    // FormatVolume formats a volume with the provided file system
+    // FormatVolume formats a volume with NTFS.
     rpc FormatVolume(FormatVolumeRequest) returns (FormatVolumeResponse) {}
-    // ResizeVolume performs resizing of the partition and file system for a block based volume
+    // ResizeVolume performs resizing of the partition and file system for a block based volume.
     rpc ResizeVolume(ResizeVolumeRequest) returns (ResizeVolumeResponse) {}
-    // VolumeStats gathers DiskSize, VolumeSize and VolumeUsedSize for a volume
-    rpc VolumeStats(VolumeStatsRequest) returns (VolumeStatsResponse) {}
-    // GetVolumeDiskNumber gets the disk number of the disk where the volume is located
-    rpc GetVolumeDiskNumber(VolumeDiskNumberRequest) returns (VolumeDiskNumberResponse) {}
-    // GetVolumeIDFromMount gets the volume id for a given mount
-    rpc GetVolumeIDFromMount(VolumeIDFromMountRequest) returns (VolumeIDFromMountResponse) {}
-    // WriteVolumeCache write volume cache to disk
+    // GetVolumeStats gathers total types and used bytes for a volume.
+    rpc GetVolumeStats(GetVolumeStatsRequest) returns (GetVolumeStatsResponse) {}
+    // GetDiskNumberFromVolumeID gets the disk number of the disk where the volume is located.
+    rpc GetDiskNumberFromVolumeID(GetDiskNumberFromVolumeIDRequest) returns (GetDiskNumberFromVolumeIDResponse ) {}
+    // GetVolumeIDFromTargetPath gets the volume id for a given mount.
+    rpc GetVolumeIDFromTargetPath(GetVolumeIDFromTargetPathRequest) returns (GetVolumeIDFromTargetPathResponse) {}
+    // WriteVolumeCache write volume cache to disk.
     rpc WriteVolumeCache(WriteVolumeCacheRequest) returns (WriteVolumeCacheResponse) {}
 }
 
 message ListVolumesOnDiskRequest {
-    // Disk device ID of the disk to query for volumes
-    string disk_id = 1;
+    // Disk device number of the disk to query for volumes.
+    uint32 disk_number = 1;
+    // The partition number.
+    uint32 partition_number = 2;
```

Comment:
mention partition number is optional. By default, it uses the first partition on the disk

Reply:
proto3 doesn't have a way to mark this field as optional, I've added the comment that it's optional though